core/src/main/java/io/questdb/std/Unsafe.java

```diff
@@ -50,9 +56,11 @@ public final class Unsafe {
     //#endif
     private static final AnonymousClassDefiner anonymousClassDefiner;
     private static final LongAdder[] COUNTERS = new LongAdder[MemoryTag.SIZE];
+    public static final ConcurrentMap<Long, StackTraceElement[]> ALLOCATION_MAPS;
 
     static {
         try {
```

Comment:
I'm not sure if using assertions is the right thing because we do have them enabled in questdb.sh and systemd.service  so some users could get this mechanism enabled on update .

Reply:
yes, this draft PR is older than my knowledge of having assertions enabled in the product. clearly this is too expensive to have it enabled. I might hide it behind another switch/guard or remove it from the PR altogether. do you have any other idea?

Reply:
I think guarding leak code with a final boolean flag (set using configuration setting or runtime flag) shouldn't be too expensive .

Reply:
agree